web_widget_image_paint/static/src/js/web_paint.js

```diff
@@ -0,0 +1,607 @@
+            if (file) {
+                var reader = new FileReader();
+                reader.readAsDataURL(file);
+                reader.onload = function (e) {
+                    self.__reset_with_local_file = e.target.result;
+                    self.__is_reset = true;
+                    self.__is_reset_with_local_file = true;
+                    self.render_value();
+                }
+            }
+        },
+        // **********************************************************
+        // * reset_clicked (custom)
+        // **********************************************************
+        reset_clicked: function (ev) {
+            this.__is_reset = true;
+            this.render_value();
+        },
+        // **********************************************************
+        // * get_url
+        // **********************************************************
+        get_url: function (background_image_field) {
+            var self = this;
+            var background_url = self.attrs['background'] || self.placeholder;
+            var url = background_url;
+            if (background_image_field) {
+                var image_field = self.fields[background_image_field];
+                if (image_field) {
+                    url = session.url('/web/image', {
+                        model: self.model,
+                        field: self.nodeOptions.preview_image ? self.nodeOptions.preview_image : background_image_field,
+                        unique: field_utils.format.datetime(self.recordData.__last_update).replace(/[^0-9]/g, ''),
+                        id: JSON.stringify(self.recordData.id || null),
+                    });
+                };
+            };
+            if (self.__is_reset && !self.get('value')) {
+                if (self.__is_reset_with_local_file) {
+                    self.__is_reset_with_local_file = false;
+                    url = self.__reset_with_local_file;
+                }
+                self.__is_reset = false;
+            }
+            else {
+                if (self['value']) {
+                    url = session.url('/web/image', {
+                        model: self.model,
+                        field: self.nodeOptions.preview_image ? self.nodeOptions.preview_image : self.name,
+                        unique: field_utils.format.datetime(self.recordData.__last_update).replace(/[^0-9]/g, ''),
+                        id: JSON.stringify(self.recordData.id || null),
+                    });
+                }
+            };
+            return url || background_url;
+        },
+        // **********************************************************
+        // * get_options
+        // **********************************************************
+        get_options: function () {
+            var self = this;
+            var options = {};
+            var options = self.attrs.options;
+            if (options['drawing_mode'] == undefined) {
```

Comment:
If you want to set default values to options why don't you do it using `_.defaults`? Search for its usage in `web`.